src/lib/spinel/spinel_driver.hpp

```diff
@@ -0,0 +1,253 @@
+ *     derived from this software without specific prior written permission.
+ *
+ *  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ *  AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ *  IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+ *  ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+ *  LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+ *  CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+ *  SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+ *  INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+ *  CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+ *  ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+ *  POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#ifndef SPINEL_DRIVER_HPP_
+#define SPINEL_DRIVER_HPP_
+
+#include <openthread/instance.h>
+
+#include "common/array.hpp"
+#include "common/callback.hpp"
+#include "lib/spinel/logger.hpp"
+#include "lib/spinel/spinel.h"
+#include "lib/spinel/spinel_interface.hpp"
+
+namespace ot {
+namespace Spinel {
+
+/**
+ * Maximum number of Spinel Interface IDs.
+ *
+ */
+#if OPENTHREAD_CONFIG_MULTIPAN_RCP_ENABLE
+static constexpr uint8_t kSpinelHeaderMaxNumIid = 4;
+#else
+static constexpr uint8_t kSpinelHeaderMaxNumIid = 1;
+#endif
+
+typedef void (
+    *ReceivedFrameHandler)(const uint8_t *aFrame, uint16_t aLength, uint8_t aHeader, bool &aSave, void *aContext);
+typedef void (*SavedFrameHandler)(const uint8_t *aFrame, uint16_t aLength, void *aContext);
+
+class SpinelDriver : public Logger
+{
+public:
+    /**
+     * Constructor of the SpinelDriver.
+     *
+     */
+    SpinelDriver(void);
+
+    /**
+     * Initialize this SpinelDriver Instance.
+     *
+     * @param[in]  aSpinelInterface            A reference to the Spinel interface.
+     * @param[in]  aSoftwareReset              TRUE to reset on init, FALSE to not reset on init.
+     * @param[in]  aIidList                    A Pointer to the list of IIDs to receive spinel frame from.
+     *                                         First entry must be the IID of the Host Application.
+     * @param[in]  aIidListLength              The Length of the @p aIidList.
+     *
+     */
+    void Init(SpinelInterface    &aSpinelInterface,
+              bool                aSoftwareReset,
+              const spinel_iid_t *aIidList,
+              uint8_t             aIidListLength);
+
+    /**
+     * Deinitialize this SpinelDriver Instance.
+     *
+     */
+    void Deinit(void);
+
+    /**
+     * Clear the rx frame buffer.
+     *
+     */
+    void ClearRxBuffer(void) { mRxFrameBuffer.Clear(); }
+
+    /**
+     * Set the internal state of co-processor as ready.
+     *
+     * This method is used to skip a reset.
+     */
+    void SetCoProcessorReady(void) { mIsCoProcessorReady = true; }
```

Comment:
It seems both `CoProcessor` and `Coprocessor` are used.

```suggestion
    void SetCoprocessorReady(void) { mIsCoProcessorReady = true; }
```

Reply:
Let's use `Coprocessor`. Have updated.